### PR TITLE
Docs: Update documentation about installing from Helm Chart

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- ~~Bugfix~~
- ~~Enhancement / new feature~~
- ~~Refactoring~~
- Documentation

### Description

This PR updted the documentation section about installing using Helm chart. This section should be upstream only.

### CODE REVIEW

File: documentation/book/con-product-downloads.adoc [hunk 1]
```diff
@@ -8,6 +8,6 @@ include::common/attributes.adoc[]
 = {ProductName} Downloads
 
 {ProductName} releases are available for download from {ReleaseDownload}.
-The release artifacts contain documentation and example YAML files for deployment on {ProductPlatformName}.
+The release artifacts contain documentation, example YAML files for deployment on {ProductPlatformName} and Helm Chart for deployment using https://helm.sh/[Helm^].
```
Comment (tombentley) [inline documentation/book/con-product-downloads.adoc]: ...and a Helm Chart for deployment...

File: documentation/book/proc-deploying-cluster-operator-helm-chart.adoc [hunk 2]
```diff
@@ -0,0 +1,28 @@
+// Module included in the following assemblies:
+//
+// assembly-cluster-operator.adoc
+
+[id='deploying-cluster-operator-helm-chart-{context}']
+= Deploying the Cluster Operator using Helm Chart
+
+.Prerequisites
+
+* Helm has to be installed on the local machine
```
Comment (tombentley) [inline documentation/book/proc-deploying-cluster-operator-helm-chart.adoc]: Each bullet should end with a full stop

File: documentation/book/proc-deploying-cluster-operator-helm-chart.adoc [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+. Verify whether the Cluster Operator has been deployed successfully using the Helm command line tool:
++
+[source]
+----
+helm ls
+----
+
+.Additional resources
+* For more information about Helm, see https://helm.sh/[Helm website^]
```
Comment (tombentley) [inline documentation/book/proc-deploying-cluster-operator-helm-chart.adoc]: see the Helm website

File: documentation/book/con-product-downloads.adoc [hunk 4]
```diff
@@ -8,6 +8,6 @@ include::common/attributes.adoc[]
 = {ProductName} Downloads
 
 {ProductName} releases are available for download from {ReleaseDownload}.
-The release artifacts contain documentation and example YAML files for deployment on {ProductPlatformName}.
+The release artifacts contain documentation, example YAML files for deployment on {ProductPlatformName} and a Helm Chart for deployment using https://helm.sh/[Helm^].
```
Comment (dmuntima) [inline documentation/book/con-product-downloads.adoc]: - Add comma before "and" (on {ProductPlatformName} and a ...)
- Add "link:" before the hyperlink (see Styling Links guidelines: http://strimzi.io/contributing/guide/#styling_links)

File: documentation/book/proc-deploying-cluster-operator-helm-chart.adoc [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+* Helm has to be installed on the local machine.
+* Helm has to be installed in the {ProductPlatformName} cluster.
+
+.Procedure
+
+. Deploy the Cluster Operator using the Helm command line tool:
++
+[source,shell,subs=attributes+]
+helm install {ChartReleaseDownload}
```
Comment (dmuntima) [inline documentation/book/proc-deploying-cluster-operator-helm-chart.adoc]: Change "{ChartReleaseDownload}" to "<_ChartReleaseDownload_>". Replaceable text has to styled in monospace and surrounded by angle brackets.

http://strimzi.io/docs/master/#document-conventions-str

Comment (scholzj) [reply documentation/book/proc-deploying-cluster-operator-helm-chart.adoc]: This is actually not something the user should replace. This is a value from attributes.,

**Pull request merged**